pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -1266,22 +1215,9 @@ func (pl *DynamicResources) bindClaim(ctx context.Context, state *stateData, ind
 	// Patch the pod status with the new information about the generated
 	// special resource claim.
 	if isExtendedResourceClaim {
-		var cer []v1.ContainerExtendedResourceRequest
-		if nodeAllocation, ok := state.nodeAllocations[nodeName]; ok {
-			cer = nodeAllocation.containerResourceRequestMappings
-		}
-		if len(cer) == 0 {
-			return nil, fmt.Errorf("nil or empty request mappings, no update of pod %s/%s ExtendedResourceClaimStatus", pod.Namespace, pod.Name)
-		}
-
-		podStatusCopy := pod.Status.DeepCopy()
-		podStatusCopy.ExtendedResourceClaimStatus = &v1.PodExtendedResourceClaimStatus{
-			RequestMappings:   cer,
-			ResourceClaimName: claim.Name,
-		}
-		err := schedutil.PatchPodStatus(ctx, pl.clientset, pod.Name, pod.Namespace, &pod.Status, podStatusCopy)
+		err := pl.patchPodExtendedResourceClaimStatus(ctx, pod, claim, nodeName, state)
 		if err != nil {
-			return nil, fmt.Errorf("update pod %s/%s ExtendedResourceClaimStatus: %w", pod.Namespace, pod.Name, err)
+			return nil, err
```

Comment:
curious, why returning err only, instead of "update pod %s/%s ExtendedResourceClaimStatus: %w"

Reply:
i see you moved it over the the function.